packages/aws-cdk-lib/aws-ecs/lib/container-definition.ts

```diff
@@ -384,6 +384,42 @@ export interface ContainerDefinitionOptions {
    * An array of ulimits to set in the container.
    */
   readonly ulimits?: Ulimit[];
+
+  /**
+   * Enable a restart policy for a container.
+   *
+   * When you set up a restart policy, Amazon ECS can restart the container without needing to replace the task.
+   *
+   * @default false
+   * @see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-restart-policy.html
+   */
+  readonly enableRestartPolicy?: boolean;
```

Comment:
Instead of having an interface called `RestartPolicy`, the flag and other properties are flatly defined.

There are two reasons for this.

The first is to comply with [the CDK design guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat).

Secondly, it was uncomfortable that an empty `RestartPolicy` had to be specified to enable the restart policy with the default configuration.

```ts
taskDefinition.addContainer('container', {
  image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
  restartPolicy: {},
});
```

I also implemented it before using the interface `RestartPolicy`. See the following diff.

https://github.com/aws/aws-cdk/pull/31228/commits/66a3db5facee0c2b31f78ece6d9f3d98eb56e957